src/passes/bytesConverter.ts

```diff
@@ -0,0 +1,152 @@
+  }
+
+  visitBinaryOperation(node: BinaryOperation, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
+      node.typeString = replacementTypeNode.pp();
+    }
+    this.commonVisit(node, ast);
+  }
+
+  visitFunctionCall(node: FunctionCall, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
+      node.typeString = replacementTypeNode.pp();
+    }
+    this.commonVisit(node, ast);
+  }
+
+  visitIdentifier(node: Identifier, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
+      node.typeString = replacementTypeNode.pp();
+    } else if (node.vReferencedDeclaration instanceof VariableDeclaration) {
+      const referencedTypeNode = getNodeType(node.vReferencedDeclaration, ast.compilerVersion);
+      if (referencedTypeNode instanceof ArrayType) {
+        if (node.vReferencedDeclaration.storageLocation === DataLocation.Default)
+          node.typeString = `${referencedTypeNode.pp()} storage ref`;
+        else if (node.vReferencedDeclaration.storageLocation === DataLocation.Memory)
+          node.typeString = `${referencedTypeNode.pp()} memory`;
+      }
+    } else if (node.vReferencedDeclaration instanceof FunctionDefinition) {
+      // Visit FunctionDefinition to ensure variable declarations for bytesN have
+      // been replaced with uintN.
+      this.commonVisit(node.vReferencedDeclaration, ast);
+      node.typeString = getFunctionTypeString(node.vReferencedDeclaration, ast.compilerVersion);
+    }
+  }
+
+  visitIndexAccess(node: IndexAccess, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
+      node.typeString = replacementTypeNode.pp();
+    }
+    this.commonVisit(node, ast);
+  }
+
+  visitLiteral(node: Literal, _: AST): void {
+    if (node.typeString.startsWith('int_const ') && node.value.startsWith('0x')) {
+      node.value = this.replacementLiteralIntValue(node);
+    }
+  }
+
+  visitMemberAccess(node: MemberAccess, ast: AST): void {
+    const typeNode = getNodeType(node, ast.compilerVersion);
+    if (typeNode instanceof FixedBytesType) {
+      const replacementTypeNode = this.replacementIntTypeNode(typeNode);
+      node.typeString = replacementTypeNode.pp();
+    } else if (
+      node.vExpression instanceof Identifier &&
+      node.vExpression.vReferencedDeclaration instanceof VariableDeclaration &&
+      node.vExpression.vReferencedDeclaration.vType instanceof ArrayTypeName &&
+      node.memberName === 'push'
```

Comment:
Why the push function in particular? Fixed-size bytes types don't have a push function, and dynamically sized bytes types will have to handled completely differently

Reply:
This was meant to handle pushes to an array of fixed-size bytes types. Although transpilation still succeeds if this typeString is not changed, I thought it would be cleaner to do so. What do you think?

```
      | +-FunctionCall #93
      |   |   typeString: tuple()
      |   +-MemberAccess #91
      |   | |   memberName: push
      |   | |   typeString: function (bytes2)
      |   | +-Identifier #89
      |   |       name: d
      |   |       referencedDeclaration: 8
      |   |       typeString: bytes2[] storage ref
      |   +-Literal #92
      |         typeString: int_const 4660
      |         value: 0x1234
```